File: angular-java-springboot/server/src/main/java/com/lcaohoanq/pfolio/services/ProfileService.java
package com.lcaohoanq.pfolio.services;

import com.lcaohoanq.pfolio.models.Profile;
import com.lcaohoanq.pfolio.models.SocialLink;
import org.springframework.stereotype.Service;

import java.util.Arrays;

@Service
public class ProfileService implements IProfileService{

    @Override
    public Profile getProfile() {
        Profile profile = Profile.builder()
            .name("Luu Cao Hoang")
            .avatarUrl("/assets/avatar.jpeg")
            .githubAccount("GitHub account is lcaohoanq")
            .role("Fullstack Web Developer")
            .build();

        SocialLink github = SocialLink.builder()
            .type("github")
            .url("https://github.com/lcaohoanq")
            .iconUrl("/assets/github.png")
            .build();

        SocialLink instagram = SocialLink.builder()
            .type("ig")
            .url("https://instagram.com/lcaohoanq")
            .iconUrl("/assets/ig.png")
            .build();

        profile.setSocialLinks(Arrays.asList(github, instagram));
        
        return profile;
    }
}

File: kotlin-springboot-thymeleaf/src/main/kotlin/com/lcaohoanq/pfolio/demo/domain/profile/ProfileService.kt
package com.lcaohoanq.pfolio.demo.domain.profile

import com.lcaohoanq.pfolio.demo.domain.social.SocialLink
import org.springframework.stereotype.Service

@Service
class ProfileService : IProfileService {

    override fun getProfile(): Profile {
        var profile = Profile(
            name = "Luu Cao Hoang",
            avatarUrl = "/assets/avatar.jpeg",
            githubAccount = "GitHub account is lcaohoanq",
            role = "Backend Developer",
        )

        with(profile) {
            socialLinks = listOf(
                SocialLink(
                    type = "github",
                    url = "https://github.com/lcaohoanq",
                    iconUrl = "/assets/github.png"
                ), SocialLink(
                    type = "instagram",
                    url = "https://www.instagram.com/lcaohoanq",
                    iconUrl = "/assets/ig.png"
                )
            )
        }

        return profile
    }

}